client/starwhale/base/view.py

```diff
@@ -126,6 +133,7 @@ def prepare_build_bundle(project: str, bundle_name: str, typ: str) -> URI:
             project=_project_uri.project,
             obj_type=typ,
             obj_name=bundle_name,
+            obj_ver=gen_uniq_version(),
```

Comment:
why add the obj version in the prepare build process?

when we call the function once, it will return a different version, is it expected?

Reply:
#### why add the obj version in the prepare build process?
In current pr, -o json is not implemented in bundle building functions(dataset, runtime, model, eval), I want the explicit uri be returned by the api. but the version is generated deep in the building process which is hard and verbose to populate the uri from layer to layer until to the return clause. So i put the version generation in the building preparing phase, when building is success, the uri is returned directlly.

#### when we call the function once, it will return a different version, is it expected?
since the function name is `prepare_build_bundle`, i think it is expected that a version is generated in the  preparing phase.